### ISSUE TITLE
Can't Strafe Forward/Drain with Grapple Voltage

### ISSUE THREAD

**Issue by 37KarsOnMars** [shiiion/dolphin#122]
I have a problem with the strafe function. When I lock on, I can strafe left and right and move backward. However, when I try to move forward, I can't unless I'm already moving forward when I lock on. When I stop moving forward while locking on, I can't move forward again.

This hasn't bothered me until I got the Grapple Voltage. For some reason, I can drain energy from objects but can't drain energy into objects unless I'm already moving forward when I use the Grapple Voltage. When I stop draining energy, I can't start draining energy until I do the same thing again.

I have a feeling that these two issues are related.

**Comment by mgperkow** [comment]
I'm not sure this issue was ever addressed. I just got to the point where I picked up the Grapple Voltage, and I can't find _any way_ to charge or drain energy. I grapple on the same way I do when I pull things, but the forward and reverse controls don't do anything except make it look like I'm trying to pull on the energy node. I've been fiddling with different control configurations for about half an hour now, hoping I'll stumble onto something, but it doesn't seem like there's any way to do this in PrimeHack (and thus proceed past this point in the game where I'm at now).

**Comment by SirMangler** [comment]
> I'm not sure this issue was ever addressed. I just got to the point where I picked up the Grapple Voltage, and I can't find _any way_ to charge or drain energy. I grapple on the same way I do when I pull things, but the forward and reverse controls don't do anything except make it look like I'm trying to pull on the energy node. I've been fiddling with different control configurations for about half an hour now, hoping I'll stumble onto something, but it doesn't seem like there's any way to do this in PrimeHack (and thus proceed past this point in the game where I'm at now).

If this is a bug, it's definitely new, we explicitly tested charge and drain in a few different places without any issues. This makes me wonder if its similar to an issue we had a few years back when dolphin introduced a new "combo bind" expression which cancels out other buttons, then made a dozen of them default in Hotkeys which broke quite a few of our controls. With that in mind, I would like to hear what happens when you clear your Hotkeys and default your controls. 

For what it's worth, the grapple lasso code and related hasn't really been touched since it was introduced, so a new issue with it suggests something external is causing or triggering it.

**Comment by mgperkow** [comment]
You know what? That's pretty much what ended up fixing it. Actually, I reentered a lot of my _same_ controls, and one of the times, it just started working after that. I gotta remember to do that it I run into anything else like this. But, yeah, the Grapple Lasso is working fine. If there's any bug, it's some of the controls sometimes coming unbound like that, but that of course would be a much more subtle and hard to trace problem, I'd imagine.
